mage/query-modules/python/elasticsearch-synchronization.md

```diff
@@ -0,0 +1,315 @@
+      backgroundColor: color,
+      borderRadius: '2px',
+      color: '#fff',
+      padding: '0.2rem',
+    }}>
+{children}
+</span>
+);
+
+<!--[![docs-source](https://img.shields.io/badge/source-elasticsearch_synchronization-FB6E00?logo=github&style=for-the-badge)]("") -->
+
+## Abstract
+**Elasticsearch** is a **text-processing platform** that can be used to enhance the capabilities of a graph database like Memgraph. It offers many fine-grained features useful when working on a text that is impossible to develop in databases. Data residing in Elasticsearch and Memgraph should be **synchronized** because otherwise, the whole system could be in an inconsistent state. Such a feature can be added inside Memgraph by using triggers: every time a new entity is added (node or edge) it gets indexed to the Elasticsearch index.
+
+
+### Blog Posts
+
+The following blog posts explain how Memgraph integrated Elasticsearch into its technology stack:
+<!--- [Elasticsearch](TODO add URL when blog post will be published)-->
+
+### About the query module
+
+In this module you can find support for the following interesting features:
+- creating **Elasticsearch index** from Memgraph clients using Cypher
+- **indexing all data** inside Memgraph to Elasticsearch indexes
+- managing Elasticsearch **authentication** in a secure way
+- indexing entities (nodes and edges) as they are being inserted into the database **without reindexing old data**
+- **scanning and searching documents** from Elasticsearch indexes using **Query DSL**
+- reindexing existing documents from Elasticsearch
+
+
+### Usage
+
+The following procedure is expected when using **Elasticsearch synchronization modules**:
+
+- start Elasticsearch instance and store securely **username, password, a path to the certificate file and instance's URL**
+- connect to the instance by calling the `connect` method
+- use the `create_index` method to create Elasticsearch indexes for nodes and edges
+- index all entities inside the database using the `index_db` method
+- check that documents were indexed correctly using the `scan` or `search` method
+
+
+## Procedures
+
+<RunOnSubgraph/>
+
+The module for synchronizing Elasticsearch with Memgraph is organized as a **stateful** module where it is expected that the user performs a sequence of operations using the managed secure connection to Elasticsearch. The user can **use indexes that already exist** inside Elasticsearch but can also choose **to create new ones with custom schema**. Indexing can be performed in two ways:
+1. index all data residing inside the database
+2. **incrementally index entities** as they get inserted into the database by using triggers. Find out more about triggers [here](https://memgraph.com/docs/memgraph/how-to-guides/set-up-triggers) and [here](https://memgraph.com/docs/memgraph/reference-guide/triggers) but essentially triggers offer us a way of executing a specific procedure upon receiving some event.
+
+
+### `connect()`
```

Comment:
you need arguments in function signature

Reply:
I think it is better like this because there are many arguments so it will be messy + there are already listed below